### ISSUE TITLE
Replicating Datahike, is it possible to transact datoms directly?

### ISSUE THREAD

**Issue by mrapogee** [replikativ/datahike#139]
Hello,

I'm experimenting with datahike replication via kafka. I have 2 nodes:

- A write datahike node, which transacts data, then sends + commits it's generated datoms to a  kafka compact topic before replying to the client with ok.

- A read node, which consumes the tx kafka topic 

On the read node, I'd like to skip the `dh/transact` function if possible for consistency + perf and directly commit datoms to the node. Is this possible?

**Comment by whilo** [comment]
Yes, that is not hard to do, but we do not provide an interface yet for it. We can provide it if it turns out to be the right thing to do. A disadvantage of this approach is that you still have to recreate the index on all nodes. An alternative would be to use a konserve backend on a distributed key-value system such as CouchDB, Redis or DynamoDB and then just send the current database value with the root nodes of the trees over the Kafka channel and unpack that on the other nodes. Would this approach make sense to you?

**Comment by whilo** [comment]
Or we could send all new tree nodes over Kafka and you store them locally if you do not want to use a separated distributed store.

**Comment by mrapogee** [comment]
That's a good point, would eliminate the need for setting up replication consensus if I wanted cluster-wide acid semantics. Curious what the latency hit would be vs. embedded level db, but eventually that could be optimized with an in-memory cache of heaviest hit nodes + updating the local cache via kafka.

Will take a look at the konserve backends.

**Comment by whilo** [comment]
Hey, did you make progress? It would be very interesting for us to have some Kafka experimentation documented as we plan to run Kafka internally as well soon.
